### ISSUE TITLE
NPM fails to install

### ISSUE THREAD

**Issue by paddatrapper** [jangouts/jangouts#348]
Following https://github.com/jangouts/jangouts/blob/master/DEVELOPMENT.md, `npm install` fails with the following error:

```
../src/create_string.cpp: In function ‘char* create_string(Nan::MaybeLocal<v8::Value>)’:
../src/create_string.cpp:17:37: error: no matching function for call to ‘v8::String::Utf8Value::Utf8Value(v8::Local<v8::Value>&)’
   17 |   v8::String::Utf8Value string(value);
      |                                     ^
In file included from /home/kyle/.node-gyp/12.18.3/include/node/node.h:67,
                 from ../../nan/nan.h:56,
                 from ../src/create_string.cpp:1:
/home/kyle/.node-gyp/12.18.3/include/node/v8.h:3135:5: note: candidate: ‘v8::String::Utf8Value::Utf8Value(v8::Isolate*, v8::Local<v8::Value>)’
 3135 |     Utf8Value(Isolate* isolate, Local<v8::Value> obj);
      |     ^~~~~~~~~
/home/kyle/.node-gyp/12.18.3/include/node/v8.h:3135:5: note:   candidate expects 2 arguments, 1 provided
make: *** [binding.target.mk:129: Release/obj.target/binding/src/create_string.o] Error 1
make: Leaving directory '/home/kyle/projects/jangouts/node_modules/node-sass/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/kyle/projects/jangouts/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Linux 5.7.0-3-amd64
gyp ERR! command "/usr/bin/node" "/home/kyle/projects/jangouts/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
gyp ERR! cwd /home/kyle/projects/jangouts/node_modules/node-sass
gyp ERR! node -v v12.18.3
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
Build failed with error code: 1
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-sass@3.13.1 postinstall: `node scripts/build.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-sass@3.13.1 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

Looks like this upstream issue: https://github.com/sass/node-sass/issues/2959. I can't see a direct dependency in `package.json`, so I'm not sure where in the dependency train it is.

NPM Version: 6.14.8
Node version: v12.18.3
Jangouts version: `master` @ 33ac5ff05ae77b4bf1b2114431d6558041deb5c8

**Comment by dgdavid** [comment]
Hi there!

I have no problems when doing the `npm install` with

* npm 6.14.6
* Node 10.22.0

```
[...]

make: Leaving directory '/home/dgdavid/development/jangouts/node_modules/node-sass/build'
gyp info ok 
Installed to /home/dgdavid/development/jangouts/node_modules/node-sass/vendor/linux-x64-64/binding.node
npm notice created a lockfile as package-lock.json. You should commit this file.
added 1102 packages from 616 contributors and audited 1106 packages in 180.305s
```

However, I could see a lot of warnings during the `npm install` output. I'll update my system and keep investigating.


> Looks like this upstream issue: [sass/node-sass#2959](https://github.com/sass/node-sass/issues/2959). I can't see a direct dependency in `package.json`, so I'm not sure where in the dependency train it is.
> 


[`npm-ls`](https://docs.npmjs.com/cli-commands/ls.html) can help here


```
➜ npm ls node-sass
jangouts@0.5.0 /home/dgdavid/development/jangouts-1009
└─┬ gulp-sass@2.1.1
  └── node-sass@3.13.1 
```

**Comment by dgdavid** [comment]
> 
> [`npm-ls`](https://docs.npmjs.com/cli-commands/ls.html) can help here
> 

Only after a successful `npm install`.

**Comment by dgdavid** [comment]
I have updated Node.js to v12.18.3 and I can reproduce the error. So, it seems a problem with the `node-sass` supported environment. In fact, it started supporting Node.js 12 in `node-sass v4.12.0` (see https://github.com/sass/node-sass/pull/2633), which also supports older Node.js versions (see https://github.com/sass/node-sass/releases/tag/v4.12.0).

So, updating the `gulp-sass` version to the latest available (https://github.com/dlmanning/gulp-sass/releases/tag/v4.1.0) in the _package.json_ file fix the issue since its `node-sass` dependency is `~4.8.3` which is solved to [4.14.1](https://github.com/sass/node-sass/releases/tag/v4.14.1) at the time of writing and, if I'm not wrong, it supports all known Node.js versions until now.

@ancorgs  :point_up_2:

**Comment by dgdavid** [comment]
@paddatrapper,

Just out of curiosity, Did you test it? Did it work for you?

**Comment by paddatrapper** [comment]
With that change, it installs correctly, but fails to run using `gulp serve`:

```
fs.js:36
} = primordials;
    ^

ReferenceError: primordials is not defined
    at fs.js:36:5
    at req_ (/home/kyle/projects/jangouts/node_modules/natives/index.js:143:24)
    at Object.req [as require] (/home/kyle/projects/jangouts/node_modules/natives/index.js:55:10)
    at Object.<anonymous> (/home/kyle/projects/jangouts/node_modules/vinyl-fs/node_modules/graceful-fs/fs.js:1:37)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
    at Module.load (internal/modules/cjs/loader.js:985:32)
    at Function.Module._load (internal/modules/cjs/loader.js:878:14)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at require (internal/modules/cjs/helpers.js:72:18)
```

**Comment by dgdavid** [comment]
Ok, thanks for the feedback. I'll take a look in the incoming days.

**Comment by Novienzi** [comment]
hey, how you @paddatrapper solve the error when running the gulp-serve? Thanks..

**Comment by paddatrapper** [comment]
> hey, how you @paddatrapper solve the error when running the gulp-serve? Thanks..

I didn't...

**Comment by dgdavid** [comment]
@paddatrapper,

I really sorry, I completely forget this issue until I received the Github notification today (thanks @Novienzi!). I hope you can forgive me.

I took a look now and, according to [this StackOverflow response](https://stackoverflow.com/a/55926692), it's a  versions problem. I tested to update all dependencies to their latest versions but now I'm facing a `AssertionError [ERR_ASSERTION]: Task never defined: clean` error, most probably because all Gulp files must be [updated to use the Gulp 4 sintax](https://www.joezimjs.com/javascript/complete-guide-upgrading-gulp-4/). Before doing that, I'll give a shot to [this other popular response](https://stackoverflow.com/questions/55921442/how-to-fix-referenceerror-primordials-is-not-defined-in-node/58394828#58394828), but I'm not sure if I could to it today.

If any of you do it first, please feel free to make a pull request with right changes.

**Comment by paddatrapper** [comment]
@dgdavid no problem. I have been distracted by other things too, so haven't had a chance to continue my experimentation anyway

**Comment by ancorgs** [comment]
We recently released version 0.6.0 which completely changes the development environment and dependencies (eg. no gulp anymore). So all these problems should have become obsolete.
